### ISSUE TITLE
[Client bug]: Filtering doesn't work when searching files by path (v5.7.0)

### ISSUE THREAD

**Issue by bav2212** [microsoftgraph/msgraph-sdk-dotnet#1857]
**Describe the bug**
Filtering doesn't work when searching files by path. $filter query parameter doesn't present in generated URI.

**To Reproduce**
```cs
var searchResult = await _graphServiceClient
	.Drives["driveId"]
	.Root
	.ItemWithPath("path/to/folder")
	.Children
	.GetAsync(requestConfiguration =>
	{
	   requestConfiguration.QueryParameters.Filter = "name eq 'filename'";
	});
```
generated request URI ($filter query parameter is absent in URI)
`https://graph.microsoft.com/v1.0/drives/driveId/root:/path/to/folder:/children`

When I'm doing the request with id usage, everything works as expected
```cs
var searchResult = await _graphServiceClient
	.Drives["driveId"]
	.Items["itemId"]
	.Children
	.GetAsync(requestConfiguration =>
	{
	   requestConfiguration.QueryParameters.Filter = "name eq 'filename'";
	});
```
generated URI:
`https://graph.microsoft.com/v1.0/drives/driveId/items/itemId/children?$filter=name eq %27filename%27`

**Expected behavior**
`$filter` query parameter should be added to request URI and expected URI should be:
`https://graph.microsoft.com/v1.0/drives/driveId/root:/path/to/folder:/children?$filter=name eq %27filename%27}`

**Client version**
5.7.0

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser No
 - Version No

**Comment by bav2212** [comment]
@andrueastman Looks like that's an issue inside Kiota. Should I report this issue in `kiota-abstractions-dotnet` repository?

**Comment by andrueastman** [comment]
Closing this as its resolved in latest 5.11.0 update
